core/src/main/java/org/infinispan/transaction/tm/DummyTransaction.java

```diff
@@ -45,7 +47,7 @@
    private final Xid xid;
    private volatile int status = Status.STATUS_UNKNOWN;
    private final List<Synchronization> syncs;
-   private final List<XAResource> resources;
+   private final Map<XAResource, Integer> resources;
```

Comment:
I'm a little worried about this line:
https://github.com/infinispan/infinispan/pull/3979/files#diff-9bce40d7971a6c1e61f18ac552eac0c3R196
It is possible the put to replace an existing XaResource.